pandas/core/groupby/groupby.py

```diff
@@ -1892,9 +1904,20 @@ def median(self, numeric_only: bool = False):
         Series or DataFrame
             Median of values within each group.
         """
+        if not (
+            type(self.obj).median is Series.median
+            or type(self.obj).median is DataFrame.median
+        ):
```

Comment:
if this pattern is going to show up a lot, does it merit a decorator?

Reply:
I've had a first pass at making a decorator, not sure how to deal with `engine` and `engine_kwargs`